src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs

```diff
@@ -1582,7 +1614,14 @@ private void WriteRedRewriter()
                         }
                         else
                         {
-                            WriteLine("      var {0} = ({1})this.Visit(node.{2});", CamelCase(field.Name), field.Type, field.Name);
+                            if (!IsOptional(field) && field.Type != "SyntaxToken")
+                            {
+                                WriteLine("      var {0} = ({1})this.Visit(node.{2}) ?? throw new ArgumentNullException(\"{0}\");", CamelCase(field.Name), GetFieldType(field, green: false), field.Name);
```

Comment:
>{1} [](start = 60, length = 3)

This cast ends up coming out as `?` often, which is wrong. We should not be trying to cast to a nullable reference type and throwing if it's null.

Reply:
What form would you suggest instead?

Reply:
The generated cast should not have a ? in it.

Reply:
So to clarify, you want this?

```diff
-(T?)Method() ?? throw ...
+(T)(Method() ?? throw ...)
```

Reply:
> We should not be trying to cast to a nullable reference type and throwing if it's null.

FWIW, this is exactly what we do today, you just can't see it in the code.

Reply:
That's correct. And more accurately, we cast to a regular reference type and throw if it was null.

Reply:
Since this is a cosmetic change only, I'm going to file this as a follow-up issue to clean up the generated code for this case.